res/controllers/Denon-Prime-4-scripts.js

```diff
@@ -0,0 +1,699 @@
+        midi: [0x9F, 0x0C],
+        group: "[QuickEffectRack1_[Channel1]]",
+        key: "enabled",
+        on: 0x02,
+        off: 0x01,
+        input: function(_channel, _control, value, _status, _group) {
+            for (let i = 1; i <= 4; i++) {
+                if (value > 0) {
+                    const effect = "[QuickEffectRack1_[Channel" + i + "]]";
+                    if (engine.getParameter(effect, "enabled") > 0) {
+                        engine.setParameter(effect, "enabled", 0);
+                    } else {
+                        engine.setParameter(effect, "enabled", 1);
+                    }
+                }
+            }
+        },
+    });
+
+    // Sweep FX - Echo Button
+    Prime4.sweepEcho = new components.Button({
+        midi: [0x9F, 0x0D],
+    });
+    // Sweep FX - Wash Button
+    Prime4.sweepWash = new components.Button({
+        midi: [0x9F, 0x0E],
+    });
+    // Sweep FX - Noise Button
+    Prime4.sweepNoise = new components.Button({
+        midi: [0x9F, 0x0F],
+    });
+
+    // Initial LED values to set (Hopefully these will automatically initialize, but for now they won't.)
+    midi.sendShortMsg(0x9F, 0x1C, colDeck[0]);                        // Deck 1 Toggle
+    midi.sendShortMsg(0x9F, 0x1D, colDeck[1]);                        // Deck 2 Toggle
+    midi.sendShortMsg(0x9F, 0x1E, colDeckDark[2]);                    // Deck 3 Toggle
+    midi.sendShortMsg(0x9F, 0x1F, colDeckDark[3]);                    // Deck 4 Toggle
+    midi.sendShortMsg(0x94, 0x21, colDeck[0]);                        // Left Jog Wheel
+    midi.sendShortMsg(0x95, 0x21, colDeck[1]);                        // Right Jog Wheel
+    midi.sendShortMsg(0x94, 0x0B, Prime4.rgbCode.blue);             // Hot-Cue Button - Left Deck
+    midi.sendShortMsg(0x95, 0x0B, Prime4.rgbCode.blue);             // Hot-Cue Button - Right Deck
+    midi.sendShortMsg(0x9F, 0x0B, 1);                                 // Headphone Split Button
+    midi.sendShortMsg(0x94, 0x1C, 1);                                 // Left Shift Button
+    midi.sendShortMsg(0x95, 0x1C, 1);                                 // Right Shift Button
+};
+
+Prime4.shutdown = function() {
+    // Return all LEDs to initial dim state
+    midi.sendShortMsg(0x90, 0x75, 0x01);
+};
+
+// All components contained in each mixer strip
+const mixerStrip = function(deckNumber, midiOffset) {
+    components.Deck.call(this, deckNumber);
+
+    // FX Buttons
+    this.fxBank1 = new components.Button({
+        midi: [0x90 + midiOffset, 0x01],
+        group: "[EffectRack1_EffectUnit1]",
+        key: "group_[Channel" + deckNumber + "]_enable",
+        type: components.Button.prototype.types.toggle,
+    });
+    this.fxBank2 = new components.Button({
+        midi: [0x90 + midiOffset, 0x02],
+        group: "[EffectRack1_EffectUnit2]",
+        key: "group_[Channel" + deckNumber + "]_enable",
+        type: components.Button.prototype.types.toggle,
+    });
```

Comment:
Please use the `components.EffectAssignmentButton`s instead. https://github.com/mixxxdj/mixxx/wiki/Components-JS#effectassignmentbutton

Reply:
Will do. Thanks